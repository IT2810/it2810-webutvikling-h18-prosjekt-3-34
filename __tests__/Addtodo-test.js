import React from "react";
import Addtodo from ".././components/addtodo";
import { Text, View, TouchableOpacity } from "react-native";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import Modal from "react-native-modal";
import styles from "../stylesheets/addtodo.style.js";

// Snapshot test
test("renders correctly", () => {
  const tree = renderer.create(<Addtodo />).toJSON();
  expect(tree).toMatchSnapshot();
});

//TODO: Render and render input. Maype include props in the
//Test if children components renders correctly
const shallow = new ShallowRenderer();
shallow.render(<Addtodo />);
const result = shallow.getRenderOutput();

test("Renders correct type", () => {
  expect(result.type).toBe(Modal);
});

it("ToggleTodoInput should change showTodoInputField", () => {
  let addTodoComponent = renderer
    .create(<Addtodo setType={() => {}} />) // Not interested to test setType function of App.
    .getInstance();
  expect(addTodoComponent.state.showTodoInputField).toBe(false);
  addTodoComponent.toggleTodoInput();
  expect(addTodoComponent.state.showTodoInputField).toBe(true);
  addTodoComponent.toggleTodoInput();
  expect(addTodoComponent.state.showTodoInputField).toBe(false);
});

it("ToggleStepInput should change showStepInputField", () => {
  let addTodoComponent = renderer
    .create(<Addtodo setType={() => {}} />) // Not interested to test setType function of App.
    .getInstance();
  expect(addTodoComponent.state.showStepInputField).toBe(false);
  addTodoComponent.toggleStepInput();
  expect(addTodoComponent.state.showStepInputField).toBe(true);
  addTodoComponent.toggleStepInput();
  expect(addTodoComponent.state.showStepInputField).toBe(false);
});

it("CloseModal should set all States to false", () => {
  let addTodoComponent = renderer
    .create(<Addtodo toggleModal={() => {}} setType={() => {}} />) // Not interested to test toggle modal function of App.
    .getInstance();
  expect(addTodoComponent.state.showStepInputField).toBe(false);
  expect(addTodoComponent.state.showTodoInputField).toBe(false);
  addTodoComponent.toggleTodoInput();
  addTodoComponent.toggleStepInput();
  expect(addTodoComponent.state.showStepInputField).toBe(true);
  expect(addTodoComponent.state.showTodoInputField).toBe(true);
  addTodoComponent.closeModal();
  expect(addTodoComponent.state.showStepInputField).toBe(false);
  expect(addTodoComponent.state.showTodoInputField).toBe(false);
});
