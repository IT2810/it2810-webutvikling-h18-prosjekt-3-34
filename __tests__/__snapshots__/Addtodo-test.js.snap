// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#eeeeee",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 24,
            "marginTop": 8,
          }
        }
      >
        Add a goal for selected day: 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Thursday, October 18th
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#a3b3c3",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "width": 65,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "padding": 5,
            }
          }
        >
          Todo
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#a3b3c3",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 1,
            "width": 65,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "padding": 5,
            }
          }
        >
          Steps
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#494949",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 35,
            "justifyContent": "center",
            "marginBottom": 15,
            "marginTop": 5,
            "opacity": 1,
            "padding": 5,
            "width": 55,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Exit
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
